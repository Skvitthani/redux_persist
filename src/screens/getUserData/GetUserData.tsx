import React from 'react';
import {RootState, flatListItem} from '..';
import ButtonComp from '../../components/ButtonComp';
import {useDispatch, useSelector} from 'react-redux';
import {DELETE_DATA} from '../../redux/types/Actiontype';
import {DealerNavigNavigationType} from '../../navigation';
import {FlatList, StyleSheet, Text, View} from 'react-native';
import {NativeStackScreenProps} from '@react-navigation/native-stack';

type AddUserDataScreen = NativeStackScreenProps<
  DealerNavigNavigationType,
  'GetUserData'
> & {
  navigation: () => void;
};

const GetUserData: React.FC<AddUserDataScreen> = ({navigation}) => {
  const dispatch = useDispatch();
  const Userdata = useSelector((state: RootState) => state?.user?.UserData);
  const renderItem = ({item}: {item: flatListItem}) => {
    return (
      <View style={styles.renderItemView}>
        <Text style={styles.textStyle}>{`Name:${item?.name}`}</Text>
        <Text style={styles.textStyle}>{`City:${item?.city}`}</Text>
        <Text style={styles.textStyle}>{`Age:${item?.age}`}</Text>
      </View>
    );
  };
  const onDeleteDataPress = () => {
    dispatch({
      type: DELETE_DATA,
    });
    navigation.goBack();
  };

  const ListEmptyComponent = () => {
    return (
      <View style={styles.emptyView}>
        <Text style={styles.noDataText}>No Data Found</Text>
      </View>
    );
  };

  return (
    <View style={styles.container}>
      <FlatList
        // testID="List_Data_FlatList"
        contentContainerStyle={{flex: 1}}
        data={Userdata}
        renderItem={renderItem}
        ListEmptyComponent={ListEmptyComponent}
      />
      <View style={styles.buttonView}>
        <ButtonComp
          buttonCompTestCase="delete_data_button"
          onPress={onDeleteDataPress}
          title="Delete Data"
        />
      </View>
    </View>
  );
};

export default GetUserData;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'white',
  },
  renderItemView: {
    margin: 10,
    padding: 10,
    borderRadius: 10,
    backgroundColor: '#1B4242',
  },
  textStyle: {
    color: 'white',
  },
  buttonView: {
    marginBottom: 50,
  },
  emptyView: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  noDataText: {
    fontSize: 20,
    color: 'black',
    fontWeight: 'bold',
  },
});
