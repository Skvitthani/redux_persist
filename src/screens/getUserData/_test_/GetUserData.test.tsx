import 'react-native';
import React from 'react';
import GetUserData from '../GetUserData';
import {fireEvent, render, waitFor} from '@testing-library/react-native';

const mockDispatch = jest.fn();
jest.mock('react-redux', () => ({
  useSelector: jest.fn().mockReturnValue({
    repositories: {
      loading: false,
      repos: [],
    },
  }),
  useDispatch: () => mockDispatch,
}));

const component = (
  <GetUserData
    navigation={
      {
        navigate: jest.fn(),
        goBack: jest.fn(),
      } as any
    }
    route={undefined as any}
  />
);

describe('GetUserData', () => {
  it('renders correctly', () => {
    const container = render(component);
    expect(container.toJSON()).toMatchSnapshot();
  });

  it('delete data button', () => {
    const {getByTestId} = render(component);
    const container = getByTestId('delete_data_button');
    fireEvent.press(container);
  });

  // it('FlatList render list', async () => {
  //   const {getByTestId} = render(component);
  //   await waitFor(() => getByTestId('List_Data_FlatList'));
  //   const container = getByTestId('List_Data_FlatList');
  //   expect(container).toBeDefined();
  // });
});
