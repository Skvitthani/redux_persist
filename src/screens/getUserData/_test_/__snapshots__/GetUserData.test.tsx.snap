// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GetUserData renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    ListEmptyComponent={[Function]}
    contentContainerStyle={
      {
        "flex": 1,
      }
    }
    data={
      {
        "repositories": {
          "loading": false,
          "repos": [],
        },
      }
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={0.0001}
    stickyHeaderIndices={[]}
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View
        style={
          {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            {
              "color": "black",
              "fontSize": 20,
              "fontWeight": "bold",
            }
          }
        >
          No Data Found
        </Text>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      {
        "marginBottom": 50,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#706233",
          "borderRadius": 20,
          "marginHorizontal": 30,
          "opacity": 1,
          "padding": 20,
        }
      }
      testID="delete_data_button"
    >
      <Text
        style={
          {
            "color": "white",
            "fontSize": 20,
            "fontWeight": "bold",
          }
        }
      >
        Delete Data
      </Text>
    </View>
  </View>
</View>
`;
